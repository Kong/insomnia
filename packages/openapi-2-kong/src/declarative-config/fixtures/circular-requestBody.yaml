openapi: 3.0.1

info:
  title: Testing Circular
  version: 1.1.1

  contact:
    name: Kong Insomnia

servers:
  - url: https://some.random.url

x-kong-plugin-request-validator:
  enabled: true
  version: draft4
  config:
    verbose_response: true

tags:
  - name: Testing

paths:
  /testing:
    post:
      requestBody:
        $ref: '#/components/requestBodies/TestRequest'
      responses:
        '200':
          description: success

components:
  requestBodies:
    TestRequest:
      content:
        application/json:
          schema:
            $ref: '#/components/schemas/TestRequest'

  schemas:
    TestRequest:
      allOf:
        - $ref: '#/components/schemas/CircularStart'

    CircularStart:
      allOf:
        - $ref: '#/components/schemas/CircularEnd'

    CircularEnd:
      allOf:
        - $ref: '#/components/schemas/CircularStart'
