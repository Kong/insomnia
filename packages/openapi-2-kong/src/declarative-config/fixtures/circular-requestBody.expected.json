{
  "_format_version": "1.1",
  "services": [
    {
      "host": "some.random.url",
      "name": "Testing_Circular",
      "path": "/",
      "plugins": [
        {
          "config": {
            "body_schema": "{}",
            "verbose_response": true,
            "version": "draft4"
          },
          "enabled": true,
          "name": "request-validator",
          "tags": ["OAS3_import", "OAS3file_circular-requestBody.yaml"]
        }
      ],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "methods": ["POST"],
          "name": "Testing_Circular-testing-post",
          "paths": ["/testing$"],
          "plugins": [
            {
              "config": {
                "body_schema": "{}",
                "verbose_response": true,
                "version": "draft4"
              },
              "enabled": true,
              "name": "request-validator",
              "tags": ["OAS3_import", "OAS3file_circular-requestBody.yaml"]
            }
          ],
          "strip_path": false,
          "tags": ["OAS3_import", "OAS3file_circular-requestBody.yaml"]
        }
      ],
      "tags": ["OAS3_import", "OAS3file_circular-requestBody.yaml"]
    }
  ]
}
